apps/test/unit/templates/levelSummary/SummaryTopLinks.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import {mount} from 'enzyme';
+import {expect} from '../../../util/reconfiguredChai';
+import {Provider} from 'react-redux';
+import {combineReducers, createStore} from 'redux';
+import SummaryTopLinks from '@cdo/apps/templates/levelSummary/SummaryTopLinks';
+import styles from '@cdo/apps/templates/levelSummary/check-for-understanding.module.scss';
+import isRtl from '@cdo/apps/code-studio/isRtlRedux';
+import viewAs from '@cdo/apps/code-studio/viewAsRedux';
+import teacherSections from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import progress from '@cdo/apps/code-studio/progressRedux';
+
+const JS_DATA = {
+  level: {
+    id: 0,
+  },
+};
+
+const INITIAL_STATE = {
+  isRtl: false,
+  viewAs: 'Instructor',
+  teacherSections: {
+    selectedStudents: [{id: 0}],
+    selectedSectionId: 0,
+    sectionIds: [0],
+    sections: [{id: 0, name: 'test section'}],
+  },
+  progress: {
+    currentLessonId: 0,
+    currentLevelId: '0',
+    lessons: [
+      {
+        id: 0,
+        levels: [{activeId: '0', position: 1}],
+      },
+    ],
+  },
+};
+
+const setUpWrapper = (state = {}, jsData = {}) => {
+  const store = createStore(
+    combineReducers({
```

Comment:
My preference here would be to export an `UnconnectedSummaryTopLinks` and not use redux in this test file. Personally, I think it's cleaner that way, but feel free to push back.

To expand: I'm suggested adding a link to `SummaryTopLinks.jsx` like `export const UnconnectedSummaryTopLinks = SummaryTopLinks` (possible I have my syntax wrong) then using that in this test. That was you don't have to mess around with creating the store and can instead pass props in directly.

Reply:
oh! good idea!

Reply:
re-reading this, I misunderstood it the first time. some questions, for my own understanding: is it common practice to test on the unconnected component like this? when should/shouldn't I do it? (e.g. why only `SummaryTopLinks` and not `SummaryResponses`?) Why don't we want the code inside the `connect()` to be hit by the tests too?

Reply:
we do use the "unconnected" components in tests a lot through our repo (eg [ShareDialog](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/ShareDialog.jsx) and [corresponding test](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/code-studio/components/ShareDialogTest.js). That said, I forgot there's some logic in the `connect` function so that might be worth testing. Honestly, that's a good argument to me to keep the store in the test file.